java/code/src/com/suse/manager/reactor/hardware/HardwareMapper.java

```diff
@@ -896,22 +904,28 @@ public void mapNetworkInfo(Map<String, Network.Interface> interfaces, Optional<M
         Optional<NetworkInterface> primaryNetIf = primaryIPv4.flatMap(pipv4 ->
             server.getNetworkInterfaces().stream()
                 .filter(netIf -> netIf.getIPv4Addresses().stream()
-                        .anyMatch(addr -> ObjectUtils.equals(pipv4, addr.getAddress())))
+                        .anyMatch(addr -> Objects.equals(pipv4, addr.getAddress())))
                 .findFirst());
 
-        if (!primaryNetIf.isPresent()) {
+        if (primaryNetIf.isEmpty()) {
             // no primary IPv4, fallback to IPv6
             primaryNetIf = primaryIPv6.flatMap(pipv6 ->
                 server.getNetworkInterfaces().stream()
                     .filter(netIf -> netIf.getIPv6Addresses().stream()
-                            .anyMatch(addr -> ObjectUtils.equals(pipv6, addr.getAddress())))
+                            .anyMatch(addr -> Objects.equals(pipv6, addr.getAddress())))
                     .findFirst());
         }
 
         // we found an interface with the same addr as the
         // primary IPv4/v6 addr, make it primary
         primaryNetIf.ifPresent(server::setPrimaryInterface);
 
+        // set primary FQDN to hostname if no primary FQDN is specified
+        if (StringUtils.isNotBlank(server.getHostname()) && server.getFqdns().stream()
+                .filter(ServerFQDN::isPrimary).findFirst().isEmpty()) {
```

Comment:
`filter(<condition>).findFirst().isEmpty()` should be equivalent to `noneMatch(<condition>)`

```suggestion
        if (StringUtils.isNotBlank(server.getHostname()) && server.getFqdns().stream()
               .noneMatch(ServerFQDN::isPrimary)) {
```